avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroGenerator.java

```diff
@@ -382,6 +390,17 @@ public final void writeStartObject() throws IOException {
     }
 
     @Override
+    public void writeStartObject(Object forValue) throws IOException {
```

Comment:
Implemented because we need to know the actual value type to pick the correct branch of a union